mailcore2 depends on icu.
The icu headers provided by the freedesktop sdk require c++17 to build, so bump it.
The project builds fine with c++17, so this is an easy fix.

--- a/mailsync/Vendor/mailcore2/CMakeLists.txt	2026-02-22 19:13:42.111382600 +0100
+++ b/mailsync/Vendor/mailcore2/CMakeLists.txt	2026-02-22 19:27:41.938339292 +0100
@@ -25,7 +25,7 @@
   set(mac_libraries iconv)
   
 ELSEIF(${CMAKE_SYSTEM_NAME} MATCHES "Linux")
-  set(CMAKE_CXX_FLAGS "-std=gnu++0x")
+  set(CMAKE_CXX_FLAGS "-std=gnu++17")
   
   set(additional_includes
     /usr/include/tidy
